buidl/mnemonic.py

```diff
@@ -34,2082 +38,63 @@ def secure_mnemonic(num_bits=256, extra_entropy=0):
     total = (preseed << checksum_bits_needed) | checksum
     # now we get the mnemonic passphrase
     mnemonic = []
+
     # now group into groups of 11 bits
     for _ in range((num_bits + checksum_bits_needed) // 11):
         # grab the last 11 bits
         current = total & ((1 << 11) - 1)
         # insert the correct word at the front
-        mnemonic.insert(0, WORD_LIST[current])
+        mnemonic.insert(0, get_bip39_word(current))
         # shift by 11 bits so we can move to the next set
         total >>= 11
     # return the mnemonic phrase by putting spaces between
     return " ".join(mnemonic)
 
 
-WORD_LOOKUP = {}
+BIP39_WORDS = []
+BIP39_LOOKUP = {}
 
 
-def __init__():
-    # go through every word
-    for i, word in enumerate(WORD_LIST):
-        # add the word's index in the hash WORD_LOOKUP
-        WORD_LOOKUP[word] = i
-        # if the word is more than 4 characters, also keep
-        #  a lookup of just the first 4 characters
-        if len(word) > 4:
-            WORD_LOOKUP[word[:4]] = i
+def all_bip39_words():
+    # lazy load
+    if not BIP39_WORDS:
+        __init__()
+    for word in BIP39_WORDS:
+        yield word
```

Comment:
Is there a performance impact here? I would think the performance bottleneck would be reading the `txt` file (which happens all at once), not iterating through the words.

Not a big deal.